### PR TITLE
Gracefull handling of exceptions that occurs when DevHostAgent closes sockets

### LINKED ISSUE #264: Unhandled exception when task is canceled during network read
If you stop debuging when you are receiving network packages, you will get a exception in the log files.

2023-04-27T12:03:57.8368051Z | Library | TRACE | StartReceivingAsync 8080 id 2 exception System.IO.IOException: Unable to read data from the transport connection: En befintlig anslutning tvingades att stänga av fjärrvärddatorn..\n ---> System.Net.Sockets.SocketException (10054): En befintlig anslutning tvingades att stänga av fjärrvärddatorn.\n   --- End of inner exception stack trace ---\n   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error, CancellationToken cancellationToken)\n   at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource<System.Int32>.GetResult(Int16 token)\n   at System.Threading.Tasks.ValueTask`1.ValueTaskSourceAsTask.<>c.<.cctor>b__4_0(Object state)\n--- End of stack trace from previous location ---\n   at Microsoft.BridgeToKubernetes.Common.PortForward.ReversePortForwardManager.ReversePortForwardInstance.StartReceivingAsync(TcpClient tcpClient, Int32 streamId, CancellationToken cancellation) in C:\Repos\Bridge-To-Kubernetes-GitHub\src\common\PortForward\ReversePortForwardManager.cs:line 152

This exception should probobly be handled gracefully since this exception is expected.

### PR DESCRIPTION
**Pull request by Eneuman**
This PR fixes System.Io.Exception, TaskCanceledException and SocketException that occured because of missing exception handling.

Theese exceptions occured because when a user stop debugging, DevHostAgent recives a command to stop. This triggers DevHostAgent to close the TcpClients from the server side. In the EndPointManager this behavior was not handled so it still tried to use the TcpClients causing different kinds of exceptions.

When this PR is merged, errors should no longer occure in the log files on a normal run.

Fixes #264

### CODE REVIEW

File: src/common/PortForward/ServicePortForwardManager.cs [hunk 1]
```diff
@@ -171,7 +171,28 @@ private async Task RunLoopAsync(TcpClient connection, CancellationToken cancella
...
+                                cRead = await stream.ReadAsync(buffer, requestProcessingCancellationTokenSource.Token);
+                            }
+                            catch (IOException ex)
+                            {
+                                if (ex.InnerException is OperationCanceledException)
+                                {
+                                    // Cancellation requested
+                                    break;
+                                }
+                                if (ex.InnerException is SocketException se && se.SocketErrorCode == SocketError.ConnectionReset || se.SocketErrorCode == SocketError.OperationAborted)
```
Comment (hsubramanianaks) [inline src/common/PortForward/ServicePortForwardManager.cs:186]: @Eneuman there is build error , can you fix it please ? following runs have failed

https://github.com/Azure/Bridge-To-Kubernetes/actions/runs/5389595610/jobs/9783765958

```/home/runner/work/Bridge-To-Kubernetes/Bridge-To-Kubernetes/src/common/PortForward/ServicePortForwardManager.cs(186,133): error CS0165: Use of unassigned local variable 'se' [/home/runner/work/Bridge-To-Kubernetes/Bridge-To-Kubernetes/src/common/common.csproj::TargetFramework=net6.0]```

Comment (Eneuman) [reply src/common/PortForward/ServicePortForwardManager.cs]: @hsubramanianaks Fixed!

**Review by hsubramanianaks — APPROVED** [review]
looks good to me.

**Comment by cxznmhdcxz** [comment]
test passed

**Pull request merged**